Description: Don't use atexit handlers
 gnome-settings-daemon crashes due to the atexit installed in ORBit,
 which is unloaded as soon as the gconf g-s-d plugin is unloaded.
 So when main() quits, the atexit function from ORBit is not available
 anymore
Author: Rodrigo Moya <rodrigo.moya@canonical.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=645704
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/orbit2/+bug/788710
diff --git a/src/orb/orb-core/corba-orb.c b/src/orb/orb-core/corba-orb.c
index 2b2229a..4979a89 100644
--- a/src/orb/orb-core/corba-orb.c
+++ b/src/orb/orb-core/corba-orb.c
@@ -289,7 +289,11 @@ static gboolean  atexit_shutdown = FALSE;
  *   This is neccessary to clean up any remaining UDS
  * and to flush any remaining oneway traffic in buffers.
  */
+#ifdef __GNUC__
+void __attribute__ ((destructor))
+#else
 static void
+#endif
 shutdown_orb (void)
 {
 	CORBA_ORB orb;
@@ -459,8 +463,10 @@ CORBA_ORB_init (int *argc, char **argv,
 	 * tried to contact them when Evolution was started the next
 	 * time, causing it to hang, too.
 	 */
+#ifndef __GNUC__
 	g_atexit (shutdown_orb);
 #endif
+#endif
 
 	retval->default_giop_version = GIOP_LATEST;
 
